src/BoolUnification.js

```diff
@@ -27,6 +27,17 @@ export function boolUnify(f1, f2) {
         throw new Error(`Illegal argument 'y': ${f2}.`)
     }
 
+    // Special case for when one side of the equation is a variable. This will generate a
+    // 'simpler' substitution than SVE, especially when the equation side has lots of variables.
+    if (isSingleVarSpecialCase(f1, f2)) {
+        let subst = {[f1.name]: f2}
+        return {status: "success", subst: subst}
+    }
+    if (isSingleVarSpecialCase(f2, f1)) {
+        let subst = {[f2.name]: f1}
+        return {status: "success", subst: subst}
+    }
```

Comment:
Are we sure of the soundness of this optimization? I talked to Jaco. He proposed something slightly different (which is clearly sound, if less efficient): If there is a variable `x` on the LHS that is not on the RHS then pick that variable as the *first* variable to use in SVE. (I forget if this works with the way we compute the free variables.).

Reply:
Figured I'd respond to this one before implementing a change to it.

All three suggestions should be equivalent, although I haven't proved it formally yet. The proof strategy I'm looking at currently would be induction over the list of free variables in the RHS. Indeed, I believe SVE should generate almost syntactically the same unifiers as this special case, but only after minimization, which starts breaking down after the equation contains more than 4 variables.

Consider the following, which works the way you suggest even on current master because `a` is lexicographically prior to all other variables and thus the first variable the current SVE implementation looks at:

```
a = b || c || d || e || f || g
    yields [
        a --> (((((((((c ∨ (d ∨ (e ∨ f))) ∨ g) ∧ ¬((c ∨ (d ∨ (e ∨ f))) ∨ g)) ∨ b) ∨ c) ∨ d) ∨ e) ∨ f) ∨ g) ∨ (a ∧ (((((((((c ∨ (d ∨ (e ∨ f))) ∨ g) ∧ ¬((c ∨ (d ∨ (e ∨ f))) ∨ g)) ∨ b) ∨ c) ∨ d) ∨ e) ∨ f) ∨ g));
        b --> (((c ∨ (d ∨ (e ∨ f))) ∨ g) ∧ ¬((c ∨ (d ∨ (e ∨ f))) ∨ g)) ∨ b;
        c --> c;
        ...]
```

It's pretty clear that `b`'s mapping minimizes to just `b`, and after a little work I believe `a` should minimize to just `b || c || d || e || f || g`. So while using the LHS variable in SVE first should generate equivalent unifiers, it still requires some amount of minimization which eludes the current minimizer.

Maybe instead of this special case, the Boolean ring normal form you suggested is the general way to do this? I saw a method to compute that form in the Baader/Nipkow book and it could be interesting.

Reply:
Regardless, I have removed it from this PR as it exists on the other one, and these should be independent efforts.